### PR TITLE
Enable Perl::Critic RequireCheckedOpen Policy

### PR DESCRIPTION
**Pull request by nick-prater**

### CODE REVIEW

**Comment by ehuelsmann** [comment]
@nick-prater agreed with @sbts ; maybe adding the OS error (`$!`) helps to clarify/debug?

File: lib/LedgerSMB/File.pm [hunk 1]
```diff
@@ -247,9 +247,12 @@ sub get_for_template{
...
-                      args => [$args->{ref_key}, $args->{file_class}]
-     );
+        funcname => 'file__get_for_template',
+        args => [
+            $args->{ref_key},
+            $args->{file_class},
+        ],
+    );
     if ( -d $LedgerSMB::Sysconfig::tempdir . '/' . $$){
         die 'directory exists';
```
Comment (sbts) [inline lib/LedgerSMB/File.pm]: perhaps we can expand this error to include the dirname that exists

Comment (nick-prater) [inline lib/LedgerSMB/File.pm]: I'd rather refactor this and the associated cleanup to use File::Temp, but for now will expand the error.

File: lib/LedgerSMB/Template/DB.pm [hunk 2]
```diff
@@ -139,7 +139,8 @@ sub get_from_file {
     $fname = $2;
     my ($template_name, $format) = split /\./, $fname;
     my $content = '';
-    open my $fh, '<', $path;
+    open my $fh, '<', $path
+        or die "Failed to open template path";
```
Comment (sbts) [inline lib/LedgerSMB/Template/DB.pm]: can we include the file name ($path) in the die?

**Pull request merged**